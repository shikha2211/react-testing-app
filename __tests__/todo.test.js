import { render ,screen , cleanup } from "@testing-library/react";
import Todo from "../components/todo";
import React from 'react';
import renderer from 'react-test-renderer';

afterEach(() =>{
   cleanup
})

/* test("render div component",() => {
   const div=document.createElement("div")
   ReactDOM.render(<Todo/> , div);
}) */

test("test the div element of todo", () => {
    const todo = {id: 1, title: 'wash dishes', completed: true};
   render(<Todo todo={todo} />);
   const todoElement = screen.getByTestId("todo-1")
   expect(todoElement).toBeInTheDocument();
   expect(todoElement).toHaveTextContent("wash dishes");
   expect(todoElement).toContainHTML('</strike>');
  
})

test("test the div element of todo", () => {
   const todo = {id: 2, title: 'make dinner', completed: false};
  render(<Todo todo={todo} />);
  const todoElement = screen.getByTestId("todo-2")
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("make dinner");
  expect(todoElement).not.toContainHTML('<strike>');
})

test("create a snapshot" ,() => {
   const todo = {id: 1, title: 'wash dishes', completed: true};
   const tree = renderer.create(<Todo todo={todo} />).toJSON();
   //console.log(tree);
   expect(tree).toMatchSnapshot();
})
