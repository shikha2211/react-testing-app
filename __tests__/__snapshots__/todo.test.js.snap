// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a snapshot 1`] = `
<div
  data-testid="todo-1"
>
  <strike>
    <h1>
      todo - 
      wash dishes
    </h1>
  </strike>
</div>
`;
